modules/basegl/src/processors/imageprocessing/imagesubsetgl.cpp

```diff
@@ -0,0 +1,239 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2013-2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/processors/imageprocessing/imagesubsetgl.h>
+#include <inviwo/core/datastructures/buffer/buffer.h>
+#include <inviwo/core/interaction/events/gestureevent.h>
+#include <inviwo/core/interaction/events/mouseevent.h>
+#include <inviwo/core/interaction/events/touchevent.h>
+#include <inviwo/core/network/networklock.h>
+
+#include <modules/opengl/geometry/meshgl.h>
+#include <modules/opengl/openglutils.h>
+#include <modules/opengl/texture/textureutils.h>
+
+namespace inviwo {
+
+const ProcessorInfo ImageSubsetGL::processorInfo_{
+    "org.inviwo.ImageSubsetGL",  // Class identifier
+    "Image Subset",              // Display name
+    "Image Operation",           // Category
+    CodeState::Stable,           // Code state
+    Tags::GL,                    // Tags
+};
+const ProcessorInfo ImageSubsetGL::getProcessorInfo() const { return processorInfo_; }
+
+ImageSubsetGL::ImageSubsetGL()
+    : Processor()
+    , inport_("inputImage")
+    , outport_("outputImage")
+    , rangeX_("rangeX", "X Slices", 0, 256, 0, 256, 1, 1)
+    , rangeY_("rangeY", "Y Slices", 0, 256, 0, 256, 1, 1)
+    , shader_("img_texturequad.vert", "img_copy.frag") {
+    addPort(inport_);
+    addPort(outport_);
+    addProperties(rangeX_, rangeY_);
+    dims_ = size2_t(1, 1);
+
+    inport_.setOutportDeterminesSize(true);
```

Comment:
These can go into the constructor of the ports

Reply:
done